__tests__/components/ResourceCounts.test.tsx

```diff
@@ -0,0 +1,70 @@
+import { render, screen, act, within } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import {
+  mantineRecoilWrap,
+  getMockFetchImplementation,
+  getMockFetchImplementationError,
+} from "../helpers/testHelpers";
+import {
+  ResourceCountResponse,
+  ResourceCounts,
+  sortResourceArray,
+} from "../../components/ResourceCounts";
+
+const RESOURCE_COUNTS_BODY: ResourceCountResponse = {
+  Account: 0,
+  Appointment: 0,
+  Measure: 5,
+  Patient: 2,
+};
+
+describe("resource Counts render", () => {
+  beforeAll(() => {
+    global.fetch = getMockFetchImplementation(RESOURCE_COUNTS_BODY);
+  });
+
+  it("should display all resources and counts fetched by mock fetch", async () => {
+    await act(async () => {
+      render(mantineRecoilWrap(<ResourceCounts />));
+    });
+
+    //Expect badge component with data-testid=resourceType to contain correct resource count
+    expect(within(screen.getByTestId("Patient")).getByText("2")).toBeInTheDocument();
+    expect(screen.getByText("Patient")).toBeInTheDocument();
```

Comment:
The "TestId" way of finding page elements should be the last resort. Component tests should focus on testing by "acting as a user". In practice, this means looking at things in a way similar to how an accessible screen reader would. For these buttons we are looking for an item that has the accessible role 'button' that has name "Patient 2". Here is how to do that using the `getByRole` for the first check:

```suggestion
    // Expect buttons with correct resource counts
    expect(screen.getByRole("button", { name: "Patient 2" })).toBeInTheDocument();
```